src/compiler/binder.ts

```diff
@@ -3260,6 +3260,7 @@ namespace ts {
                 if (isInJSFile(node) &&
                     isVariableDeclarationInitializedToBareOrAccessedRequire(possibleVariableDecl) &&
                     !getJSDocTypeTag(node) &&
+                    !(lookupSymbolForName(container, "require" as __String)) &&

```

Comment:
we should move this in enum as we are doing in below

Reply:
Are you talking about an enum like ModifierFlags? `require` isn't a modifier like `public` et al, and I think if the binder gets this right, then the checker won't have to look up `require` again. Even if it does, it'll only happen once more. I don't think it's worth caching.